src/variable.ts

```diff
@@ -33,8 +45,89 @@ function plugin({
   return {
     postcssPlugin: 'postcss-variable-rename',
     prepare() {
-      // TODO(jiramide): add variable renaming logic
-      return {};
+      if (strategy === 'none' && !outputMapCallback && !prefix) return {};
+
+      const outputMap: RenamingMap | null = outputMapCallback ? {} : null;
+      const alreadyProcessedNodes = new Set<Declaration>();
+
+      const skip: SkipPredicate = createSkipPredicate(except);
+      const rename: RenamingFunction = createStrategy(strategy, skip);
+
+      /**
+       * Renames the given `variable` if prefixed with '--'; otherwise does nothing.
+       * @param variable - CSS variable to rename; this can be prefixed or not.
+       * @returns string - The renamed CSS variable. This is prefixed with '--'.
+       */
+      function renameVariable(variable: string): string {
+        if (!variable.startsWith('--')) {
+          return variable;
+        }
+
+        const strippedVariable = stripVariable(variable);
+
+        if (skip(strippedVariable)) {
+          return variable;
+        }
+
+        const renamedVariable = prefix
+          ? prefix + '-' + rename(strippedVariable)
+          : rename(strippedVariable);
+
+        if (outputMap) {
+          outputMap[strippedVariable] = renamedVariable;
+        }
+
+        return '--' + renamedVariable;
+      }
+
+      /**
+       * Walks the given AST node and renames any variables it spots as it goes
+       * NOTE: This function does not explicitly check for `var(...)`
+       * function-nodes, and it indiscriminately renames any word-type nodes
+       * it sees that has the custom CSS property prefix. As of May 30, 2025,
+       * nothing in the CSS spec suggests that this does not work, but this
+       * may break in the future in case a new CSS feature is added that
+       * reuses the double-dash prefix in a new way.
+       * @param node - The root of the AST to walk through
+       */
+      function walk(node: valueParser.Node): void {
```

Comment:
If I'm understanding correctly you only need to walk declarations right? So why not use `walkDecls`?